modules/perinatal/controller/carnet-perinatal.events.ts

```diff
@@ -0,0 +1,75 @@
+let dataLog: any = new Object(userScheduler);
+dataLog.body = { _id: null };
+dataLog.method = null;
+
+// Al validar la prestacion de vacunacion, inserta la fecha de vacunacion en la inscripcion
+EventCore.on('perinatal:control:validacion', async ({ prestacion }) => {
+    let embarazo = prestacion.ejecucion.registros.find(registro => registro.concepto.conceptId === '366321006');
+    if (embarazo) {
+        const gesta = embarazo.valor.term;
+        const carnetExistente = await CarnetPerinatalCtr.findOne({ idPaciente: prestacion.paciente.id, embarazo: gesta });
+        if (carnetExistente) {
+            if (!carnetExistente.controles) {
+                carnetExistente.controles = [];
+            }
+            let indexControl = carnetExistente.controles.findIndex(item => String(item.idPrestacion) === String(prestacion.id));
+            if (indexControl < 0) {
+                carnetExistente.controles.push({ fechaControl: prestacion.ejecucion.fecha, idPrestacion: prestacion.id });
+            }
+            carnetExistente.controles = carnetExistente.controles.sort((a, b) => {
+                return new Date(a.fechaControl).getTime() - new Date(b.fechaControl).getTime();
+            });
+            carnetExistente.fechaUltimoControl = prestacion.ejecucion.fecha;
+            await CarnetPerinatalCtr.update(carnetExistente.id, carnetExistente, dataLog);
+        } else {
+            let fechaUltimaMenstruacion = prestacion.ejecucion.registros.find(registro => registro.concepto.conceptId === '21840007');
+            let pesoPrevio = prestacion.ejecucion.registros.find(registro => registro.concepto.conceptId === '248351003');
+            let talla = prestacion.ejecucion.registros.find(registro => registro.concepto.conceptId === '14456009');
+            let fechaProbableDeParto = prestacion.ejecucion.registros.find(registro => registro.concepto.conceptId === '161714006');
+            const carnet: any = {
+                fecha: prestacion.ejecucion.fecha,
+                paciente: prestacion.paciente,
+                controles: [
+                    {
+                        fechaControl: prestacion.ejecucion.fecha,
+                        idPrestacion: prestacion.id
+                    }],
+                fechaUltimoControl: moment(prestacion.ejecucion.fecha).startOf('day').toDate(),
+                embarazo: gesta
+            };
+            if (fechaUltimaMenstruacion) {
+                carnet.fechaUltimaMenstruacion = moment(fechaUltimaMenstruacion.valor).startOf('day').toDate();
+            }
+            if (pesoPrevio) {
+                carnet.pesoPrevio = pesoPrevio.valor;
+            }
+            if (talla) {
+                carnet.talla = talla.valor;
+            }
+            if (fechaProbableDeParto) {
+                carnet.fechaProbableDeParto = moment(fechaProbableDeParto.valor).startOf('day').toDate();
+            }
+            await CarnetPerinatalCtr.create(carnet, dataLog);
```

Comment:
Existe un metodo para extraer el usuario que audito la prestación para no tener que inventar datos. 

```suggestion
    const user = Auth.getUserFromResource(prestacion);
    await CarnetPerinatalCtr.create(carnet, user);
```